src/base/rss/rssdownloadrule.cpp

```diff
@@ -50,37 +50,101 @@ DownloadRule::DownloadRule()
 
 bool DownloadRule::matches(const QString &articleTitle) const
 {
+    qDebug() << "Matching article:" << articleTitle;
+
+    qDebug() << "Checking matching tokens:" << m_mustContain.join("|");
+    bool foundMustContain = false;
+
     foreach (const QString &token, m_mustContain) {
-        if (!token.isEmpty()) {
-            QRegExp reg(token, Qt::CaseInsensitive, m_useRegex ? QRegExp::RegExp : QRegExp::Wildcard);
-            if (reg.indexIn(articleTitle) < 0)
-                return false;
+        if (token.isEmpty()) {
+            continue;
+        }
+
+        if (m_useRegex) {
+            QRegExp reg(token, Qt::CaseInsensitive, QRegExp::RegExp);
+
+            if (reg.indexIn(articleTitle) > -1) {
+                foundMustContain = true;
+            }
+        }
+        else {
+            // Split the token on spaces, then check if all of the elements are found (in any order) - if so, it's a potential match.
+            foundMustContain = true;
+
+            // We need to call replace, so can't be const &
+            foreach (QString wildToken, token.split(" ")) {
+                QRegExp reg(wildToken.replace(" ", ""), Qt::CaseInsensitive, QRegExp::Wildcard);
+
+                if (reg.indexIn(articleTitle) == -1) {
+                    foundMustContain = false;
+                    break;
+                }
+            }
         }
+
+        if (foundMustContain) {
+            qDebug() << "Found matching token: " << token;
+            break;
+        }
+    }
+
+    if (!foundMustContain) {
+        return false;
     }
-    qDebug("Checking not matching tokens");
-    // Checking not matching
+
+    qDebug() << "Checking not matching tokens:" << m_mustNotContain.join("|");
+
     foreach (const QString &token, m_mustNotContain) {
-        if (!token.isEmpty()) {
-            QRegExp reg(token, Qt::CaseInsensitive, m_useRegex ? QRegExp::RegExp : QRegExp::Wildcard);
-            if (reg.indexIn(articleTitle) > -1)
+        if (token.isEmpty()) {
+            continue;
+        }
+
+        if (m_useRegex) {
+            QRegExp reg(token, Qt::CaseInsensitive, QRegExp::RegExp);
+
+            if (reg.indexIn(articleTitle) > -1) {
+                qDebug() << "Found not matching token: " << token;
                 return false;
+            }
+        }
+
+        // Split the token on spaces, then check if all of the elements are found (in any order) - if so, reject it.
```

Comment:
The comment seems to be in contradiction with the code below: article is rejected if any of the tokens was not found.